metricflow/dag/id_prefix.py

```diff
@@ -1,14 +1,39 @@
 from __future__ import annotations
 
-from enum import Enum
+from abc import ABC, ABCMeta, abstractmethod
+from dataclasses import dataclass
+from enum import Enum, EnumMeta
 
+from typing_extensions import override
 
-class IdPrefix(Enum):
-    """Enumerates the prefixes used for generating IDs.
 
-    TODO: Move all ID prefixes here.
+class IdPrefix(ABC):
+    """A prefix for generating IDs. e.g. prefix=foo -> id=foo_0."""
+
+    @property
+    @abstractmethod
+    def str_value(self) -> str:
+        """Return the string value of this ID prefix."""
+        raise NotImplementedError
+
+
+class EnumMetaClassHelper(ABCMeta, EnumMeta):
+    """Metaclass to allow subclassing of IdPrefix / Enum.
+
+    Without this, you'll get an error:
+
+        Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all
+        its bases
+
+    since Enum has a special metaclass.
     """
 
+    pass
+
+
+class StaticIdPrefix(IdPrefix, Enum, metaclass=EnumMetaClassHelper):
```

Comment:
Oh I see.

The other option is to make IdPrefix a protocol, but we've been over the tradeoffs there and I think having it be a superclass makes sense even though enum-type subclasses are a bit weird.